plugins/acpi-dmar/fu-self-test.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupd.h>
+
+#include "fu-common.h"
+
+#include "fu-acpi-dmar.h"
+
+static void
+fu_acpi_dmar_opt_in_func (void)
+{
+	g_autoptr(FuAcpiDmar) dmar = NULL;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GBytes) blob = NULL;
+	g_autofree gchar *fn = NULL;
+
+	fn = g_build_filename (TESTDATADIR, "DMAR", NULL);
+	blob = fu_common_get_contents_bytes (fn, &error);
+	g_assert_no_error (error);
+	g_assert_nonnull (blob);
+	dmar = fu_acpi_dmar_new (blob, &error);
+	g_assert_no_error (error);
+	g_assert_nonnull (dmar);
+	g_assert_true (fu_acpi_dmar_get_opt_in (dmar));
```

Comment:
Perhaps also worth having a case that this fails for testing with?

Reply:
Do you have anything you know has a DMAR which isn't opt-in?

Reply:
Unfortunately yes.

Reply:
pushed it

Reply:
Thanks.